src/commands/ssh/startSsh.ts

```diff
@@ -0,0 +1,99 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { SiteConfigResource, User } from 'azure-arm-website/lib/models';
+import * as portfinder from 'portfinder';
+import * as vscode from 'vscode';
+import { SiteClient, TunnelProxy } from 'vscode-azureappservice';
+import { SiteTreeItem } from '../../explorer/SiteTreeItem';
+import { WebAppTreeItem } from '../../explorer/WebAppTreeItem';
+import { ext } from '../../extensionVariables';
+import { delay } from '../../utils/delay';
+import * as remoteDebug from '../remoteDebug/remoteDebugCommon';
+
+export type sshTerminal = {
+    running: boolean,
+    terminal: vscode.Terminal | undefined
+};
+
+export const sshSessionsMap: Map<string, sshTerminal> = new Map();
+
+export async function startSsh(node?: SiteTreeItem): Promise<void> {
+    if (!node) {
+        node = <SiteTreeItem>await ext.tree.showTreeItemPicker(WebAppTreeItem.contextValue);
+    }
+
+    const currentSshTerminal: sshTerminal | undefined = sshSessionsMap.get(node.root.client.fullName);
+    if (currentSshTerminal && currentSshTerminal.running) {
+        throw new Error(`Azure SSH is currently starting or already started for "${node.root.client.fullName}".`);
+    }
+
+    try {
+        sshSessionsMap.set(node.root.client.fullName, { running: true, terminal: undefined });
+        await startSshInternal(node);
+    } catch (error) {
+        sshSessionsMap.set(node.root.client.fullName, { running: false, terminal: undefined });
+        throw error;
+    }
+}
+
+async function startSshInternal(node: SiteTreeItem): Promise<void> {
+    const siteClient: SiteClient = node.root.client;
+    const siteConfig: SiteConfigResource = await siteClient.getSiteConfig();
+
+    // should always be an unbound port
+    const localHostPortNumber: number = await portfinder.getPortPromise();
+    const sshPortNumber: number = 2222;
+    const confirmDisableMessage: string = 'The app configuration will be updated to disable remote debugging and restarted. Would you like to continue?';
```

Comment:
How about `You must disable remote debugging in order to ssh. This will restart the app.` with a `Disable` button

I just worry the existing message makes it seem like I accidentally clicked "Disable remote debugging" instead of "Start SSH"